connection.go

```diff
@@ -0,0 +1,131 @@
+package papergres
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"sort"
+	"strings"
+)
+
+// SSLMode defines all possible SSL options
+type SSLMode string
+
+const (
+	// SSLDisable - No SSL
+	SSLDisable SSLMode = "disable"
+	// SSLRequire - Always SSL, no verification
+	SSLRequire SSLMode = "require"
+	// SSLVerifyCA - Always SSL, verifies that certificate was signed by trusted CA
+	SSLVerifyCA SSLMode = "verify-ca"
+	// SSLVerifyFull - Always SSL, verifies that certificate was signed by trusted CA
+	// and server host name matches the one in the certificate
+	SSLVerifyFull SSLMode = "verify-full"
+)
+
+// Connection holds all database connection configuration.
+type Connection struct {
+	Database    string
+	User        string
+	Password    string
+	Host        string
+	Port        string
+	AppName     string
+	Timeout     int
+	SSLMode     SSLMode
+	SSLCert     string
+	SSLKey      string
+	SSLRootCert string
+}
+
+// NewConnection creates and returns the Connection object to the postgres server
+func NewConnection(databaseURL string, appName string) Connection {
+	u, err := url.Parse(databaseURL)
+	if err != nil {
+		panic(err)
+	}
+
+	host, port, _ := net.SplitHostPort(u.Host)
+	p, _ := u.User.Password()
```

Comment:
It's not an error, it's a boolean value that tells whether the value has been set or not. 
If the value hasn't been set then the value returned will be defaulted as well and in this case the `p` variable will then contain an empty password. This is intentional, for scenarios where there is no user and password set for postgres. For example, when we are developing we really don't need password. 

Thus, the following postgres URIs will work:
```
postgres://{Host}
postgres://{Host}:{Port}
postgres://{Host}/{Database}
```
Since we never know beforehand if a password is required or not to connect to a DB, it's safe to ignore the bool.
Because:
-  if we check for the boolean and error out then the above ways to connect will stop working as we'll always prompt for a password.
- If there is a password required and we send "" then the actual function which establishes connection will fail to connect, which will then propagate the correct error and prompt for password.

Hence, it's safe to ignore the boolean value.